src/app/people/ConnectedCalendarEvents.tsx

```
@@ -111,168 +143,308 @@ function getEventColor(event: CalendarEvent, calendarColor?: string): string {
   }, [events]);
 
-  // Get the most recent PAST event for the disclaimer (not future events)
+  // Map date string (YYYY-MM-DD) -> events on that day (for grid highlights + tooltips)
+  const eventsByDate = useMemo(() => {
+    const map = new Map<string, Array<{ event: CalendarEvent; startDate: Date }>>();
+    filteredEvents.forEach(({ event, startDate }) => {
+      const key = format(startDate, 'yyyy-MM-dd');
+      if (!map.has(key)) map.set(key, []);
+      map.get(key)!.push({ event, startDate });
+    });
+    map.forEach((list) => list.sort((a, b) => a.startDate.getTime() - b.startDate.getTime()));
```

Biome (2.3.14)

[error] This callback passed to forEach() iterable method should not return a value.

Either remove this return or remove the returned value.


(lint/suspicious/useIterableCallbackReturn)